code/unclassified/src/maximum_subarray_sum/maxsubarraysum.cpp

```diff
@@ -0,0 +1,41 @@
+// Part of Cosmos by OpenGenus Foundation
+
+
+// C++ implementation of simple algorithm to find
+// Maximum Subarray Sum in a given array
+// this implementation is done using Kadane's Algorithm which has a time complexity of O(n)
+
+#include <iostream>
+#include <vector>
+#include <climits>
+
+int maxSubarraySum(std::vector<int> arr)
```

Comment:
```suggestion
int maxSubarraySum(const std::vector<int>& arr)
```

Prevents unnecessary copy.